### PR TITLE
Support Fuel-Async Related Libraries

### PR DESCRIPTION
**Pull request by iNoles**
## Description

Moving Corountines to Fuel-Async with Reactor and RxJava 3

## Type of change

Check all that apply

- [ ] Bug fix (a non-breaking change which fixes an issue)
- [ ] New feature (a non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Refactoring (a change which changes the current internal or external interface)
- [ ] This change requires a documentation update

## How Has This Been Tested?

In case you did not include tests describe why you and how you have verified the
changes, with instructions so we can reproduce. If you have added comprehensive
tests for your changes, you may omit this section.

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation, if necessary
- [x] My changes generate no new compiler warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Inspect the bytecode viewer, including reasoning why

### CODE REVIEW

File: fuel-async/fuel-rx3/src/main/kotlin/fuel/RxJavaExtension.kt [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package fuel
+
+import io.reactivex.rxjava3.core.Single
+import okhttp3.Call
+import okhttp3.Response
+import java.io.IOException
+
+public fun Call.toSingle(): Single<Response> =
+    Single.defer {
```
Comment (kittinunf) [inline fuel-async/fuel-rx3/src/main/kotlin/fuel/RxJavaExtension.kt:9]: [q] I wonder whether we should favor `create` over defer so we can call `cancel` properly with `call.cancel()`

**Comment by iNoles** [comment]
@SleeplessByte What is a best way to handle RxJava + OkHttp with cancel?

**Comment by SleeplessByte** [comment]
Probably expose the cancellation via a public function, and calling it automatically on dispose.

**Review by kittinunf — APPROVED** [review]
Looks good :)

**Comment by iNoles** [comment]
@kittinunf It is not ready because Codecov said missing checks on `onSuccess`

**Comment by kittinunf** [comment]
Done! I push a fix for you.

File: fuel-async/fuel-reactor/src/main/kotlin/fuel/ReactorExtension.kt [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+import okhttp3.Call
+import okhttp3.Callback
+import okhttp3.Response
+import reactor.core.publisher.Mono
+import java.io.IOException
+
+public fun Call.toMono(): Mono<Response> =
+    Mono.create {
+        enqueue(object : Callback {
```
Comment (kittinunf) [inline fuel-async/fuel-reactor/src/main/kotlin/fuel/ReactorExtension.kt:11]: Hmm I think we should prefer `execute` over `enqueue`

Comment (iNoles) [reply fuel-async/fuel-reactor/src/main/kotlin/fuel/ReactorExtension.kt:11]: I kinda surprised to see it works. :)

**Pull request merged**